### ISSUE TITLE
[tune][sgd] Add common metrics to TorchTrainable

### ISSUE THREAD

**Issue by crypdick** [ray-project/ray#7964]
I am migrating my code to RaySGD, which is incompatible with [Pytorch Ignite metrics.](https://pytorch.org/ignite/metrics.html#complete-list-of-metrics). It would be nice for sgd.torch.TorchTrainer's evaluator to have out-of-the-box support for metrics besides accuracy and loss: recall, precision, Fbeta, Dice coefficient, IoU (as well as per-class variants for multihot encodings). Ignite also makes it easy to compute confusion matrix variants.

**Comment by anabranch** [comment]
Thanks for the feedback!

cc @richardliaw

**Comment by richardliaw** [comment]
@crypdick thanks for making this issue! Do you only want this during validation?

**Comment by crypdick** [comment]
@richardliaw pretty much-- the only training set metrics I usually log are running average loss.

**Comment by valiantljk** [comment]
seems related to https://github.com/ray-project/ray/issues/16182
